.github/workflows/check-odo.yml

```diff
@@ -13,33 +13,59 @@ jobs:
     steps:
       - name: Check Out Code
         uses: actions/checkout@v4
-      - name: Get latest ODO version
+      - name: Get latest ODO nightly build version
         run: |
           echo "REPO_ODO_VERSION=$(cat src/tools.json | jq -r .odo.version)" >> $GITHUB_ENV
-          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName,url)
-          echo "LATEST_TOOL_RELEASE=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName | sed 's|v||')" >> $GITHUB_ENV
-          echo "LATEST_TOOL_URL=$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .url)" >> $GITHUB_ENV
+          LATEST_TOOL_RELEASE_RESP=$(gh release --repo ${{ env.TOOL_REPO }} view --json tagName)
+          latestReleaseTag="$(echo ${LATEST_TOOL_RELEASE_RESP} | jq -r .tagName)"
+          echo "LATEST_TOOL_RELEASE_TAG=${latestReleaseTag}" >> $GITHUB_ENV
+          echo "LATEST_TOOL_RELEASE_VERSION=$(echo ${latestReleaseTag} | sed 's|v||')" >> $GITHUB_ENV
+          commitId="$(git ls-remote --refs --heads https://github.com/${{ env.TOOL_REPO }} refs/heads/main 2>/dev/null | cut -c1-9 2>/dev/null)"
+          echo "LATEST_TOOL_COMMIT_ID=${commitId}" >> $GITHUB_ENV
+          echo "LATEST_TOOL_NIGHTLY_BUILD=${commitId}-nightly" >> $GITHUB_ENV
       - name: Find existing PR for ODO version
         run: |
-          echo PR_EXISTS=$(gh pr --repo ${{ github.repository }} list --state all --search "update odo ${{env.LATEST_TOOL_RELEASE}} in:title" --json url | jq length) >> $GITHUB_ENV
+          latestReleaseVersionString="${{ env.LATEST_TOOL_RELEASE_VERSION }} (${{ env.LATEST_TOOL_NIGHTLY_BUILD }})"
+          echo "LATEST_TOOL_RELEASE_VERSION_STRING=${latestReleaseVersionString}" >> $GITHUB_ENV
```

Comment:
Same thing here. We can just set `LATEST_TOOL_RELEASE_VERSION_STRING` directly.

I though it was weird to include the previous release version in the string that includes that this is a nightly release, but apparently this is what the `odo` nightly binary actually does :

```
$ ./odo-linux-amd64 version
===============================================
⚠  unable to fetch the cluster server version
===============================================
odo v3.15.0 (6d83e4043-nightly)

Podman Client: 4.9.4
```

Reply:
Actually it's not the "previous" it's the current release version.  
You may think of nightly build as a Service Release, so the version stays the same, just with a few bug fixes added.

Also if I don't include it - the version will look like "6d83e4043-nightly" which is hard to associate with the real  ODO version.

Reply:
> It's just odd because most projects would adopt the convention of naming it (${lastReelase}+1) since that's what is actually being targetted.

So, having `3.15.0` released, do you mean I have to enumerate the following nightly builds as `3.15.1` (xxxxxxxx-nightly)?...

I don't thing it'll properly describe the version of nightly build, because when user executes such a nightly build he'll get: `odo v3.15.0 (6d83e4043-nightly)`, not `odo v3.15.1 (6d83e4043-nightly)`

Reply:
The nightly version is changed to be in `<Last Release Version + 0.0.1>-<Build Number>-nightly`, so, f.i., for the release `3.15.0` and the commit with ID `6d83e4043` the nightly version will be generated as `3.15.1-116-nightly`, which is fully `semver` compatible and when comparing:

`3.15.0` < `3.15.1-116-nightly` < `3.15.1`